src/rulesets/oas3/index.json

```diff
@@ -78,6 +78,111 @@
         }
       }
     },
+    "valid-oas-example-in-parameters": {
```

Comment:
Whoah, whoah, whoah. Why so many rules.
We could create a custom function and utilize `functionOptions`.

Reply:
@philsturgeon, I only added rules, like you said - no new functions. But maybe again I'm mistaken? cc @P0lip

Reply:
We could have a `valid-oas-example` rule and simply instrument it using options, like so
```yaml
# your ruleset
valid-oas-example: [error, {
  in: [parameters, headers, content] # etc
}]
```

Reply:
sorry @P0lip, I don't quite follow, where ^ would go?

Reply:
@P0lip there are a lot of different ways in which examples can show up, so we were trying to get them done sharing the same generic schemaPath function, as the only difference between most of the types of differences are example and schema are siblings, or example is nested inside schema, or both, and that's just parameters. 

This PR ignores Media Type Object `example` or `examples` as explained elsewhere.

@P0lip this PR has been going on for a long time and people have wandered in and started asking lots of questions, then we've explained it, then we have started the process over again with another person. You should definitely get caught up but doing it all over text would take a fairly long time. If you two can have a call (or I can jump in too tomorrow) then it might be easiest. 

```
      parameters:
        - in: query
          name: petId
          required: true
          description: The id of the pet to retrieve
          schema:
            type: integer
            example: "3"
          example: "5"
```

Check the tests to see all sorts of examples of parameters in different places. 

If you think it'll be drastically better for performance we can switch approaches another time and make `osOas3ParameterExamples` look for `example` in parameter objects  and `osOas3SchemaExamples` that will go through all schema's that have examples nested anywhere in them or whatever. 

There's a million ways to do this we just gotta pick one that works and that isn't slow. The performance issues were discovered after this approach was decided upon.

Reply:
>  The performance issues were discovered after this approach was decided upon.

I have known for at least a month or so. I might have not communicated that though, so it's on me.

Yeah, I understand the background behind this PR and I and don't really want to prevent this PR from being merged. I am aware lots of folks are waiting for that. On the other hand, we are introducing plenty of new rules that will be disabled in Studio anyway until the performance thingy is completed, which, by the way, might require these rules to be rewritten anyway.

Let's sync tomorrow in the morning or so.
I believe a custom function based approach shouldn't take longer than half of a day / day, so we should be fine after all. Either way, as I said, I won't block this PR.
If you all really want to ship it, I'm fine with it. I'll just create a tech debt ticket to be addressed in 4.2 or 5.0 in such a case.